autoload/zepl.vim

```diff
@@ -100,6 +100,46 @@ function! zepl#send(text) abort
     endif
 endfunction
 
+function! zepl#send_sigint() abort
+  if has('nvim')
+    call jobsend(getbufvar(s:repl_bufnr, '&channel'), "\<C-c>")
+  else
+    call term_sendkeys(s:repl_bufnr, "\<C-c>")
+    call term_wait(s:repl_bufnr)
+  endif
+endfunction
+
+function! zepl#send_eof() abort
+  if has('nvim')
+    call jobsend(getbufvar(s:repl_bufnr, '&channel'), "\<C-d>")
+  else
+    call term_sendkeys(s:repl_bufnr, "\<C-d>")
+    call term_wait(s:repl_bufnr)
+  endif
+endfunction
+
+function! zepl#send_newline() abort
+  if has('nvim')
+    call jobsend(getbufvar(s:repl_bufnr, '&channel'), "\<CR>")
+  else
+    call term_sendkeys(s:repl_bufnr, "\<CR>")
+    call term_wait(s:repl_bufnr)
+  endif
+endfunction
+
+function! zepl#focus() abort
+  if !s:repl_bufnr
+    call s:error('No active REPL')
+    return
+  endif
+  let windows_containing_repl = win_findbuf(s:repl_bufnr)
+  if !empty(windows_containing_repl)
+    call win_gotoid(windows_containing_repl[0])
+  else
+    exe 'sbuffer '.s:repl_bufnr
+  endif
+endfunction
```

Comment:
Yes, it's similar. One of the reasons I made a separate function was because `zepl#jump` takes parameters, and I wasn't sure what to pass in to just focus the window -- now I see that `call zepl#jump('', 0)` would do the job.

However, there is one difference -- this function always focuses the active REPL, even if it's in a different tab (which I prefer), whereas `zepl#jump` opens the REPL in a new split.

Reply:
I've changed all of the public functions so now they have optional parameters where possible; which now means that `call zepl#jump()` with no parameters will work.

`sbuffer` (and `zepl#jump`) can actually focus buffers in other tabs, but you have to modify `:help 'switchbuf'`.

```vim
set switchbuf+=usetab
call zepl#jump()
```

Reply:
Awesome, thanks. Works just as expected. Since there is only one REPL at a time, I added `usetab` to the `set switchbuf` line [in autoload/zepl.vim](https://github.com/thezeroalpha/zepl.vim/blob/75b60eb8d5b315656c4267773028e2927f67b0bc/autoload/zepl.vim#L72), because it makes more sense to me to switch to the existing REPL (even if it's in a separate tab), as opposed to opening it in a split.

Reply:
I have actually considered making that the default option before, however if for whatever reason a user doesn't want that behaviour, they would be unable to turn it off as the option is always enabled as part of zepl.vim.

I definitely need to update the documentation to make this clear.

Reply:
Another thing you could potentially do is let the user set a variable such as `g:zepl_switchbuf`. If it's set, use its value for `switchbuf`, otherwise use the zepl defaults. That way the user could have a line in their vimrc like `set g:zepl_switchbuf = &switchbuf` to use their own settings.